lib/pbench/server/cache_manager.py

```diff
@@ -150,8 +150,8 @@ def make_cache_object(dir_path: Path, path: Path) -> CacheObject:
 
     if path.is_symlink():
         ftype = CacheType.SYMLINK
+        link_path = path.readlink()
```

Comment:
Hmm. Question is whether it's possible for this to raise an exception. Although we're only handling `FileNotFoundError` and `ValueError` ... the former, at least, being presumably unlikely as this method is used in traversing an unpacked file tree.

Reply:
It _is_ possible for `readlink()` to raise an exception.  Scenarios which leap to mind:
- `path` doesn't exist (`FileNotFoundError`)
- `path` is not a symlink (`OSError`:  `Invalid argument`)
- `path` is somehow not readable (`PermissionError`)

The question is, could any of these scenarios plausibly arise in this code?  Presumably, the second cannot, as we've already checked that at line 151.  If the file doesn't exist, then `is_symlink()` seems to return `False`, so that's not a problem, either.  And, if `path` were somehow not readable (which I don't think is possible), then `is_symlink()` would throw the exception before `readlink()` gets a chance.

So, I don't think moving the line creates any new exposure.

FWIW, the handling of `FileNotFoundError` is targeting the `Path.resolve()` call, not the `readlink()` call.